### PR TITLE
Consider updated fields for Jinja2 based computed attributes

### PR DESCRIPTION
**Pull request by ogenstad**
Changes in this PR

* The computed attribute mutation didn't manage the update in a transaction so I've wrapped the save() method within a new transaction
* Changed the node mutated event to include a "fields" key in the payload (due to this I also had to modify the webhook as it gained an extra level of "indentation", we'll probably refactor the webhook completely in a not to distant future regardless)
* Use these updated fields as part of the lookup to find computed attributes in the schema. The goal is to only run the computed attribute calculation if a relevant attribute/relationship has been updated.

Note in this iteration I've only isolated the change to when we are actually running the Computed Attributes, it will save us from querying the backend with the SDK when we know that it isn't strictly required.

I view this as an initial step to improve the performance a bit. The next iteration of this would be to modify the event automations so that the triggers themselves have these fields built in. I think it would be nice to refactor the automations in general as we'll end up with potentially quite a few additional automations when this is in place. Most of the code for this was already in place since before we started using it as automations in Prefect but the filtering was ignored as we always sent in an empty list for "updated_fields".

Also I really dislike that I did with the "update_fields" parameter and the use of ujson, we have this with the webhooks too. Would like a cleaner solution.

Final note: To improve things in a similar way for the Transforms I think we really have to spend time to rewrite and enhance the GraphQLQueryAnalyzer.

### CODE REVIEW

File: backend/infrahub/events/node_action.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class NodeMutatedEvent(InfrahubBranchEvent):
     node_id: str = Field(..., description="The ID of the mutated node")
     action: MutationAction = Field(..., description="The action taken on the node")
     data: dict[str, Any] = Field(..., description="Data on modified object")
+    fields: list[str] = Field(default_factory=list, description="Fields provided in tha mutation")
```
Comment (ajtmccarty) [inline backend/infrahub/events/node_action.py]: `tha` might be a little too casual

File: backend/infrahub/graphql/mutations/computed_attribute.py [hunk 2]
```diff
@@ -92,18 +92,20 @@ async def mutate(
...
+            async with context.db.start_transaction() as dbt:
+                await target_node.save(db=dbt, fields=[str(data.attribute)])
 
-            log_data = get_log_data()
-            request_id = log_data.get("request_id", "")
+                log_data = get_log_data()
+                request_id = log_data.get("request_id", "")
 
-            graphql_payload = await target_node.to_graphql(db=context.db, filter_sensitive=True)
+                graphql_payload = await target_node.to_graphql(db=dbt, filter_sensitive=True)
```
Comment (ajtmccarty) [inline backend/infrahub/graphql/mutations/computed_attribute.py]: not sure if this matters here, but I have encountered problems where changes are not written to the database until the transaction is closed and that prevents us from getting the latest changes.
so maybe this `to_graphql` should be outside of the transaction

File: backend/infrahub/events/node_action.py (hunk 1 above)
Comment (ogenstad) [reply backend/infrahub/events/node_action.py]: In what way? That it needs another description or that we should do something completely different? The idea is to capture the name of attributes or relationships that have been modified.

Comment (ajtmccarty) [reply backend/infrahub/events/node_action.py]: it was a joke. I assume `tha` is a misspelling of `the`

Comment (ogenstad) [reply backend/infrahub/events/node_action.py]: 😘

**Review by dgarros — APPROVED** [review]

File: backend/infrahub/events/node_action.py (hunk 1 above)
Comment (ogenstad) [reply backend/infrahub/events/node_action.py]: Will fix that.

Comment (ogenstad) [reply backend/infrahub/events/node_action.py]: I've corrected this typo. I didn't see the one I made when looking at your comment in the mobile client so I didn't quite understand what you meant :)

File: backend/infrahub/graphql/mutations/computed_attribute.py (hunk 2 above)
Comment (ogenstad) [reply backend/infrahub/graphql/mutations/computed_attribute.py]: Ok! I moved this part out of the transaction.

**Pull request merged**